academy/ibc-dev/ibcmw-integrate.md

```diff
@@ -0,0 +1,177 @@
+# Integrating IBC middleware into a chain
+
+<HighlightBox type="learning">
+
+Learn how to integrate IBC middleware(s) with a base application to your chain. The following document only applies for Cosmos SDK chains.
+
+</HighlightBox>
+
+We've now seen how to develop an IBC middleware, the final step in order to use it is to integrate it into the chain. We can distinguish between a single piece of middleware or a stack of middleware and between middleware that is stateful or stateless.
+
+To integrate our middleware (stack) we must do the following in `app.go`:
+
+- If the middleware is maintaining its own state and/or processing SDK messages, then it should **create and register its SDK module with the module manager**. Make sure not to create multiple `moduleManager`s but to register all modules in the same one.
+
+    <ExpansionPanel title="Example code">
+
+  ```go
+  //app.go
+
+  // create a keeper for the stateful middleware
+  mw1Keeper := mw1.NewKeeper(storeKey1)
+  mw3Keeper := mw3.NewKeeper(storeKey3)
+
+  // instantiate the middleware as IBCModules
+  mw1IBCModule := mw1.NewIBCModule(mw1Keeper)
+  mw2IBCModule := mw2.NewIBCModule() // optional: middleware2 is stateless middleware
+  mw3IBCModule := mw3.NewIBCModule(mw3Keeper)
+
+  // register middleware in app module
+  // if the module maintains independent state and/or processes sdk.Msgs
+  app.moduleManager = module.NewManager(
+      ...
+      mw1.NewAppModule(mw1Keeper),
+      mw3.NewAppModule(mw3Keeper),
+      transfer.NewAppModule(transferKeeper),
+      custom.NewAppModule(customKeeper)
+  )
+
+  ```
+
+    </ExpansionPanel>
+
+- As mentioned in the previous section, in the case where the middleware wishes to send a packet or acknowledgment without the involvement of the underlying application, it should be given access to the same `scopedKeeper` as the base application so that it can retrieve the capabilities by itself.
+
+    <ExpansionPanel title="Example code">
+
+  ```go
+      //app.go
+
+      // add scoped keepers in case the middleware wishes to
+      // send a packet or acknowledgment without
+      // the involvement of the underlying application
+      scopedKeeperTransfer := capabilityKeeper.NewScopedKeeper("transfer")
+      scopedKeeperCustom1 := capabilityKeeper.NewScopedKeeper("custom1")
+      scopedKeeperCustom2 := capabilityKeeper.NewScopedKeeper("custom2")
+
+
+  ```
+
+    </ExpansionPanel>
+
+  For example, if the middleware `mw1` needed the ability to send a packet without the underlying application `custom2`, it would have to get access to the latter's `scopedKeeper`:
+
+  ```diff
+  - mw1Keeper := mw1.NewKeeper(storeKey1)
+  + mw1Keeper := mw1.NewKeeper(storeKey1, scopedKeeperCustom2)
+  ```
```

Comment:
@AdityaSripal  is this correct (and sufficient)? Having trouble finding adequate info on this